homeassistant/components/thermostat/honeywell.py

```diff
@@ -52,20 +52,52 @@ def _setup_round(username, password, config, add_devices):
         return False
 
 
+def _setup_us(username, password, config, add_devices):
+    session = requests.Session()
+    if not HoneywellUSThermostat.do_login(session, username, password):
+        _LOGGER.error('Failed to login to honeywell account %s', username)
+        return False
+
+    thermostats = HoneywellUSThermostat.get_devices(session)
+    if not thermostats:
+        _LOGGER.error('No thermostats found in account %s', username)
+        return False
+
+    thermostat_id = config.get('id')
+    if thermostat_id is not None:
+        if thermostat_id not in thermostats:
+            _LOGGER.error(
+                'Requested thermostat %i not available in account %s',
+                thermostat_id, username)
+            return False
+        else:
+            thermostats = {thermostat_id: thermostats[thermostat_id]}
+
+    add_devices([HoneywellUSThermostat(id_, username, password,
+                                       name=name,
+                                       session=session)
+                 for id_, name in thermostats.items()])
+
+
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """ Sets up the honeywel thermostat. """
     username = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
-    thermostat_id = config.get('id')
+    # At one point 'id' meant 'us', so try not to break existing configs
```

Comment:
It's fine to break existing configs as the previous config has never been officially released.